File: leetcode/Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        n = len(nums)

        for i in range(n):
            sub = target - nums[i]
            if sub in m:
                return [m[sub], i]
            m[nums[i]] = i
        return []

File: leetcode/Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        int n = nums.size();

        for (int i = 0; i < n; i++){
            int sub = target - nums[i];
            if (map.count(sub)){
                return {map[sub], i};
            }
            map[nums[i]] = i;
        }
    return {};
    }
};